### PR TITLE
bazel-watcher: init at 0.5.0

### PR DESCRIPTION
**Pull request by kalbasit**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

/cc @mboes @Profpatsch @xeji

### CODE REVIEW

File: pkgs/development/tools/bazel-watcher/default.nix [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+    preInstall = ''
+      # Remove all built in external workspaces, Bazel will recreate them when building
+      rm -rf $bazelOut/external/{bazel_tools,\@bazel_tools.marker}
+      rm -rf $bazelOut/external/{embedded_jdk,\@embedded_jdk.marker}
+      rm -rf $bazelOut/external/{go_sdk,\@go_sdk.marker}
+      rm -rf $bazelOut/external/{local_*,\@local_*}
+    '';
+
+    sha256 = "0m7sjlbhg3p3l3mhs5smd28mj3zyzf6jrk3dgg0mjfb3zwfnfz32";
```
Comment (srhb) [inline pkgs/development/tools/bazel-watcher/default.nix]: If you're sure this hash was correct at the time, this derivation is not stable. I immediately got a hash mismatch here.

Comment (kalbasit) [reply pkgs/development/tools/bazel-watcher/default.nix]: interesting, thx for checking @srhb, probably something else needs to be removed before computing the hash.

Comment (kalbasit) [reply pkgs/development/tools/bazel-watcher/default.nix]: @srhb can you try to build it and upload the `/tmp/nix-build-bazel-watcher-2018-09-09-deps.drv-0` for me? Thank you!

File: pkgs/development/tools/bazel-watcher/default.nix [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  nativeBuildInputs = [ go git ];
+
+  bazelTarget = "//ibazel";
+
+  fetchAttrs = {
+    preBuild = ''
+      patchShebangs .
+
+      # tell rules_go to use the Go binary found in the system
+      sed -e 's:go_register_toolchains():go_register_toolchains(go_version = "host"):g' -i WORKSPACE
```
[the remark this answers is not in the document]
Comment (kalbasit) [reply pkgs/development/tools/bazel-watcher/default.nix]: yep, that was my comment there :) I chose to use host so I don't have to worry about matching the go version brought by `io_tweag_rules_nixpkgs` to the one in the buildInputs.

**Review by srhb — CHANGES_REQUESTED** [review]
Thanks, looks good and builds locally for me. :) A couple of things:

If this is not a released version, it should be named "bazel-watcher-unstable-2018-09-09". Is there any reason to not use a released version?

I believe the convention for the commit message is to also use the "version" as above, eg "init at 2018-09-09"

**Comment by kalbasit** [comment]
@srhb I was pointing to a specific revision because of my patch upstream https://github.com/bazelbuild/bazel-watcher/commit/4d5928eee3dd5843a1b55136d914b78fef7f25d0 that's needed for a successful packaging. But since my commit is just one after the released version v0.5.0, I have updated the PR to use the release and just import the patch.

PTAL.

File: pkgs/development/tools/bazel-watcher/default.nix (hunk 2 above)
[the remark this answers is not in the document]
Comment (kalbasit) [reply pkgs/development/tools/bazel-watcher/default.nix]: No worries :)

**Review by srhb — APPROVED** [review]
LGTM, however I'd like to understand why it fails to build on ofborg. Are there any other assumptions about the -dep bit that break CI somehow?

**Comment by kalbasit** [comment]
@srhb not sure why but it seems to fail and succeed randamly. See #46864 for my attempt at understanding the failure.

**Comment by kalbasit** [comment]
@srhb sorry I gave you the wrong PR, I meant this PR comment https://github.com/NixOS/nixpkgs/pull/46509#issuecomment-420453352

**Comment by kalbasit** [comment]
@GrahamcOfBorg build bazel-watcher

**Comment by kalbasit** [comment]
@GrahamcOfBorg build bazel-watcher

**Comment by srhb** [comment]
@grahamc Do you know what this error is about? It appears to work just fine locally, sandboxed and all. https://logs.nix.ci/?key=nixos/nixpkgs.47448&attempt_id=4a8c1c59-ccb8-424a-b8ed-ce5d6c8fe945

**Comment by kalbasit** [comment]
@grahamc also I tried restarting the build a few times in #46509 and it did end up working once. I'll give it another try.

@GrahamcOfBorg build bazel-watcher

**Comment by srhb** [comment]
The error was traced to a bug specific to nix 2.0.1. 2.0.2 or later are fine.

**Pull request merged**